packages/shared/src/components/post/SquadPostContent.tsx

```diff
@@ -0,0 +1,204 @@
+    onPreviousPost,
+    onNextPost,
+    onShare: onSharePost,
+    onClose,
+    inlineActions,
+  };
+
+  const tldrHeight = useMemo(() => {
+    if (height === null) return 'auto';
+
+    return shoudShowSummary ? height : 0;
+  }, [shoudShowSummary, height]);
+
+  const containerClass =
+    sidebarRendered && modalSizeToClassName[ModalSize.Large];
+
+  return (
+    <>
+      {position === 'fixed' && (
+        <FixedPostNavigation
+          {...navigationProps}
+          onReadArticle={onReadArticle}
+          className={className?.fixedNavigation}
+        />
+      )}
+      <PostContentContainer
+        className={classNames(
+          'relative py-8 px-6 post-content',
+          containerClass,
+          className?.container,
+        )}
+        hasNavigation={hasNavigation}
+      >
+        <BasePostContent
+          className={{
+            ...className,
+            onboarding: 'mb-6',
+            navigation: { actions: 'ml-auto', container: 'mb-6' },
+          }}
+          isLoading={isLoading}
+          isFallback={isFallback}
+          customNavigation={customNavigation}
+          enableShowShareNewComment={enableShowShareNewComment}
+          shouldOnboardAuthor={shouldOnboardAuthor}
+          navigationProps={navigationProps}
+          engagementProps={engagementActions}
+          origin={origin}
+          post={post}
+        >
+          <PostSourceInfo
+            date={postDateFormat(post.createdAt)}
+            source={post.source}
+          />
+          <span className="flex flex-row items-center mt-3">
+            <ProfileTooltip user={post.author}>
+              <ProfilePicture
+                user={post.author}
+                size="xxlarge"
+                nativeLazyLoading
+              />
+            </ProfileTooltip>
+            <ProfileTooltip
+              user={post.author}
+              link={{ href: post.author.permalink }}
+            >
+              <a className="flex flex-col ml-4">
+                <span className="font-bold">{post.author.name}</span>
+                <span className="text-theme-label-tertiary">
+                  @{post.author.username}
+                </span>
+              </a>
+            </ProfileTooltip>
+          </span>
+          <p className="mt-6 typo-title3">{post.title}</p>
+          <div className="flex flex-col mt-8 rounded-16 border border-theme-divider-tertiary">
+            <span className="flex flex-col-reverse laptop:flex-row p-4 max-w-full">
+              <div className="flex flex-col flex-1">
+                <h2 className="flex flex-wrap mt-4 laptop:mt-0 mb-4 font-bold typo-body">
+                  {post.sharedPost.title}
+                </h2>
+                <PostSourceInfo
+                  date={`${post.sharedPost.readTime}m read time`}
+                  source={post.source}
+                  typo="typo-footnote"
+                  size="small"
+                />
+                <ReadArticleButton
```

Comment:
Do we have to provide `onReadArticle` here as well?

Reply:
@el I doubled check and it is needed to provide `onReadArticle` for proper tracking